src/libs/ajax/Metrics.ts

```diff
@@ -1,32 +1,25 @@
 import { getDefaultProperties } from '@databiosphere/bard-client';
 import _ from 'lodash/fp';
 import { authOpts, fetchBard, jsonBody } from 'src/libs/ajax/ajax-common';
-import { ensureAuthSettled } from 'src/libs/auth';
 import { getConfig } from 'src/libs/config';
 import { withErrorIgnoring } from 'src/libs/error';
 import { MetricsEventName } from 'src/libs/events';
 import * as Nav from 'src/libs/nav';
-import { authStore, getSessionId } from 'src/libs/state';
-import { v4 as uuid } from 'uuid';
+import { AuthState, authStore, getSessionId } from 'src/libs/state';
 
 export const Metrics = (signal?: AbortSignal) => {
-  const captureEventFn = async (event: MetricsEventName, details: Record<string, any> = {}): Promise<void> => {
-    await ensureAuthSettled();
-    const { signInStatus, registrationStatus } = authStore.get(); // NOTE: This is intentionally read after ensureAuthSettled
-    const isRegistered = signInStatus === 'signedIn' && registrationStatus === 'registered';
-    if (!isRegistered) {
-      authStore.update(
-        _.update('anonymousId', (id) => {
-          return id || uuid();
-        })
-      );
-    }
+  const captureEventFn = async (
+    event: MetricsEventName,
+    details: Record<string, any> = {},
+    state: AuthState = authStore.get()
```

Comment:
I'm skeptical of this. Why would we ever want to pass in an auth store state? This would allow "faking" the current auth state for metrics.

Reply:
My thinking is that if we want to wait for the auth to settle, that should not be inside of metrics but rather the caller can pass in the state once auth settles

Reply:
Having `ensureAuthSettled` within captureEvent is a useful abstraction. We can call captureEvent without having to worry about the current auth state and trust that it does the right thing.

Extracting `ensureAuthSettled` places the burden of dealing with auth state on every caller. And even if `ensureAuthSettled` is extracted, there's still no need to pass in the auth state. If a caller has to wait until auth is settled before calling `captureEvent`, then `captureEvent` can still use the current state of auth store.